src/toolkit/matl_buy_policy.cc

```diff
@@ -186,11 +253,17 @@ void MatlBuyPolicy::Stop() {
 std::set<RequestPortfolio<Material>::Ptr> MatlBuyPolicy::GetMatlRequests() {
   rsrc_commods_.clear();
   std::set<RequestPortfolio<Material>::Ptr> ports;
-  bool make_req = buf_->quantity() < req_when_under_ * buf_->capacity();
+
+  // (s,S)/(R,Q) inventory policy handling
+  if (!MakeReq()) {
+    return ports;
+  }
+
   double amt;
 
   int current_time_ = manager()->context()->time();
 
+  // period inventory handling
   if (never_dormant() || current_time_ < next_active_end_) {
```

Comment:
As all this logic has become more complicated, I'm thinking about a system that separates out the different purposes into different chunks or code:
* determine if the inventory policy requires/allows a request to be made
* determine if the time requires/allows a request to be made
* update the next active/dormant times

It may cause some redundancy in the checks that occur, but it will be easier to know what is happening everyhere.

Reply:
Is this more clear? 
https://github.com/cyclus/cyclus/blob/c4ac399d2236f34cf79848a180c626d2d82fd88a/src/toolkit/matl_buy_policy.cc#L265-L307